test/common/config/http_subscription_impl_test.cc

```diff
@@ -42,6 +51,45 @@ TEST_F(HttpSubscriptionImplTest, BadJsonRecovery) {
   verifyStats(3, 1, 0, 1, 7148434200721666028);
 }
 
+TEST_F(HttpSubscriptionImplInitFetchTimeoutTest, InitialFetchTimeout) {
```

Comment:
Arguably, we should have these tests in https://github.com/envoyproxy/envoy/blob/master/test/common/config/subscription_impl_test.cc as common and provide enough machinery in the harnesses to support a common test pattern.

Reply:
I find it hard to implement these test cases in `subscription_impl_test.cc`, because we run that test against `SubscriptionType::Filesystem` too, which doesn't implement `initial_fetch_timeout`. So it is not 100% common test pattern.

Reply:
Can't we just fake the filesystem timeout behavior in the harness?

Reply:
You can also add an initial line that decides to skip that test for filesystem implementations (based on checking `GetParam()`).

Reply:
I did as you suggested. `initial_fetch_timeout` is now tested by one set of tests for all Subscription implementations. Filesystem subscription harness has fake/empty timeout behavior/expectations